### DISCUSSION TITLE
Terminal based header for mini.starter

### DISCUSSION THREAD

**Discussion by farzadmf** [nvim-mini/mini.nvim#503] [Ideas]
While looking at [this config](https://github.com/alex35mil/dotfiles/blob/master/home/.config/nvim/lua/plugins/alpha.lua#L37), it seems like `alpha-nvim` allows rendering the header section using a terminal command.

I was wondering if such functionality is achievable in `mini.starter`

Sorry if the "Ideas" section is not the proper place for this, but since I couldn't find it in the docs, I thought it could be more of an idea rather than a "general" discussion

**Comment by echasnovski** [comment]
> Sorry if the "Ideas" section is not the proper place for this, but since I couldn't find it in the docs, I thought it could be more of an idea rather than a "general" discussion

No worries, it's good.

------

Both header and footer in config [can be callables](https://github.com/echasnovski/mini.nvim/blob/25c4bf4efac51aad5393e37949e68ac3e672a73c/doc/mini-starter.txt#L224-L232). Meaning that they can evaluate script and return its output with something like this:

```lua
local header = function() return vim.fn.system('~/.config/nvim/logo.sh') end
require('mini.starter').setup({ header = header })
```

Unfortunately, this only works with the first variation (without colors), as escaped sequences are not supported.

**Reply by farzadmf** [reply]
> this only works with the first variation (without colors)

Damn, that was the main reason and motivation for this. I found it so cool to have a colorized header

**Reply by echasnovski** [reply]
Yeah, 'goolord/alpha-nvim' seems to use separate floating window for this functionality, which is a bit too much for 'mini.starter'.
